lib/src/scanner.dart

```diff
@@ -306,13 +306,13 @@ class Scanner {
   }
 
   /// Consumes the next token and returns the one after that.
-  Token advance() {
+  Token? advance() {
     scan();
     return peek();
   }
 
   /// Returns the next token without consuming it.
-  Token peek() {
+  Token? peek() {
```

Comment:
This is very unfortunate. I understand that it's because of the explicit return of `null` on line 316, but most of the code where `peek` and `advance` are called is not expecting a `null`. I'm not familiar with this package, but given that returning `null` would almost certainly have resulted in an NPE before, perhaps we should consider throwing an exception here instead of returning null. And if there are places where a `null` was expected and handled correctly before, consider adding a `peekOrNull` and `advanceOrNull` (or some other names) that do return `Token?`.

Reply:
This is not a public API, either – right?